src/lib/RpcApi.ts

```diff
@@ -147,17 +147,22 @@ export default class RpcApi {
                     hasRequest: !!this._staticStore.request,
                 });
 
+                let account;
+                if (request && 'walletId' in request) {
+                    account = await WalletStore.Instance.get((request as ParsedSimpleRequest).walletId);
+                    if (!account && requestType !== RequestType.LOGOUT) {
+                        // no account found although it's required
+                        state.reply(ResponseStatus.ERROR, new Error('Account ID not found'));
+                        return;
```

Comment:
If handling this error here, the handling can be removed in every other request component, right?

Reply:
Additionally, the error text should be made into a constant in `Constants.ts`.

Reply:
To remove it, a `!` non-null assertion needs to be added to the `walletStore.get` calls to make typescript know that it exists:
```ts
const wallet = (await WalletStore.Instance.get(this.request.walletId))!;
```
Also in weird circumstances it might still make sense, e.g. if a developer opens the popup pointing to `accounts.nimiq.com/sign-transaction` but with an onboarding request, the `SignTransaction` view can refuse to do anything due to non existing wallet.

That's why I left it in. But now that you bring it up again, I also think it should be removed.
I'll leave the error message as a string then, as it then only has one occurrence.

Reply:
Hmm, that scenario with an invalid request is a valid argument. Maybe the existance of required properties in the request should be checked in the `RequestParser.ts`?

Reply:
I removed the checks in the request views and created an issue for the request validation: https://github.com/nimiq/accounts/issues/190